projects/vdk-core/src/vdk/api/plugin/plugin_input.py

```diff
@@ -198,6 +202,63 @@ def ingest_payload(
             invocations belong to same collection. Defaults to "data_job_name|OpID",
             meaning all method invocations from a data job run will belong to the
             same collection.
+
+        :return: [Optional] IngestionResult, containing plugin-defined
+        information about the result from the ingestion operation. This result
+        could be used for post-ingestion processing operations like sending
+        telemetry data about the ingested payload.
+
+        :exceptions: Exceptions raised while ingesting data can either be
+        handled at plugin level, or they will be automatically caught at the
+        vdk-core level.
+        """
+        pass
+
+    def post_ingest_process(
+        self,
+        payload: Optional[List[dict]] = None,
+        ingestion_result: Optional[IngestionResult] = None,
+        exceptions: Optional[List] = None,
+    ) -> None:
+        """
+        Do post-ingestion processing of the ingestion payload
+        or result data from the ingestion operation.
+
+        Example Implementation for telemetry:
+
+        .. code-block:: python
+            import sys
+            import requests
+
+            def post_ingest_process(
+                self,
+                payload: Optional[List[dict]],
+                ingestion_result: Optional[IngestionResult]
+                exceptions: Optional[List]
+            ) -> None:
+                # Prepare telemetry
+                telemetry = {}
+                payload_sizes = [sys.getsizeof(i) for i in payload]
+                telemetry['payload_size'] = sum(payload_sizes)
+                telemetry['caught_ingest_exceptions'] = exceptions
+                telemetry |= ingestion_result
+
+                # Send telemetry to wherever is needed.
+                requests.post('example.com', data=telemetry)
+
+
+        :param payload: Optional[List[dict]]
+            The payloads that have been ingested. Depending on the number
+            of payloads to be processed, there might 0 or many dict objects.
+            Each dict object is a separate payload.
+        :param ingestion_result: Optional[IngestionResult]
+            The result from the ingestion operation, or some information about
+            the ingestion operation and payload that plugin developers have
+            decided to process further. This result can be either an
+            IngestionResult object (which is effectively a dictionary), or
+            None.
+        :param exceptions: Optional[List]
```

Comment:
Is this a valid use case, having a list of exceptions?

Reply:
It might be possible, but it is highly improbable. Decided to use a list just in case.